gpd/win-2/default.nix

```diff
@@ -0,0 +1,18 @@
+{ lib, pkgs, ...}:
+{
+  imports = [
+    ../../common/cpu/intel
+    ../../common/pc/laptop
+    ../../common/pc/ssd
+    ];
+
+  boot.initrd.availableKernelModules = [ "xhci_pci" "ahci" "usbhid" "usb_storage" "sd_mod" "sdhci_pci" ];
```

Comment:
For now we only extend nixos-generate-config since the needed kernel modules may change with kernel versions. This might be open for debate but than we should consistently handle this in all hardware modules in the same way.

```suggestion
```

Reply:
@Mic92 I'd really not like to remove the kernel modules here since the builtin SD card reader simply wouldn't work without them. If it's dependent on kernel versions we could set this based on kernel versioning instead?

I'm not sure what the best solution is, but removing functionality of the laptop without informing the user seems like an actually bad idea to me.

Reply:
If you run `nixos-generate-config` on the device are those not added by the script in `hardware-configuration.nix`?

Reply:
Oh, I misunderstood. If it is required that the user boots a LiveCD first and runs nixos-generate-config before using the code in this repository, then it should definitely be removed yes. I've done so.

Reply:
Ideally we could even add more auto-detection to the `nixos-generate-config` code or some similar implementation or have this script just put hardware metadata into the configuration that can be than matched against a hardware module. But until this happens this repository is a nice & easy way to put the `archlinux wiki` into code form.